add doubleClickToJoin extension

lare354 opened the pull request.

NotNite reviewed (changes requested):
Some problems with this extension:

- Correct me if I'm wrong, but this patch doesn't seem to be `require`ing your module, so it won't work properly: https://github.com/lare354/moonlight-plugins/blob/3c6c5450a658ad553e3ae78076d2afa74dc5db9c/src/doubleClickToJoin/index.tsx#L28
- Your extension doesn't have `spacepack` or `common` [set as a dependency in the manifest](https://moonlight-mod.github.io/ext-dev/manifest/#dependencies), so please fix this.

lare354 commented:
I have added the missing require() and added dependencies to manifest.json.

Support for call buttons in DMs have been added and the patches for voice and stage channels have been changed out for a better unified solution thanks to Nanakusa.
